pkg/apis/certmanager/validation/certificate_for_issuer_test.go

```diff
@@ -0,0 +1,73 @@
+package validation
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidateCertificateForACMEIssuer(t *testing.T) {
+	fldPath := field.NewPath("spec")
+	scenarios := map[string]struct {
+		spec   *v1alpha1.CertificateSpec
+		issuer *v1alpha1.IssuerSpec
+		errs   []*field.Error
+	}{
+		"valid basic certificate": {
+			spec: &v1alpha1.CertificateSpec{
+				CommonName: "testcn",
+				SecretName: "abc",
+				IssuerRef:  validIssuerRef,
+			},
+			issuer: &v1alpha1.IssuerSpec{},
```

Comment:
Why does this test case pass? If the issuer type is not defined in the IssuerSpec, surely NameForIssuer should error with `fmt.Errorf("no issuer specified for Issuer '%s/%s'", i.GetObjectMeta().Namespace, i.GetObjectMeta().Name)`?

Reply:
Ah, that is as part of `ValidateCertificateForIssuer` and not this function. Perhaps we can add a couple of test cases there too for the case where an unspecified issuer type is used?

Reply:
I've updated the test to cover `ValidateCertificateForIssuer` instead of individual validation tests